homeassistant/components/template/weather.py

```diff
@@ -452,3 +468,260 @@ def _validate_forecast(
                 )
             continue
         return result
+
+
+@dataclass(kw_only=True)
+class WeatherExtraStoredData(ExtraStoredData):
+    """Object to hold extra stored data."""
+
+    last_temperature: float | None
+    last_humidity: float | None
+    last_wind_speed: float | None
+    last_wind_bearing: float | str | None
+    last_ozone: float | None
+    last_visibility: float | None
+    last_pressure: float | None
+    last_wind_gust_speed: float | None
+    last_cloud_coverage: int | None
+    last_dew_point: float | None
+    last_apparent_temperature: float | None
+    last_forecast: list[Forecast] | None
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return a dict representation of the event data."""
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, restored: dict[str, Any]) -> Self | None:
+        """Initialize a stored event state from a dict."""
+        try:
+            return cls(  # type: ignore[misc]
+                restored["last_temperature"],
```

Comment:
The auto-generated init now only supports keywords:
```suggestion
                last_temperature=restored["last_temperature"],
```